src/sql-parser/src/ast/defs/query.rs

```diff
@@ -331,6 +390,31 @@ impl<T: AstInfo> AstDisplay for Cte<T> {
 }
 impl_display_t!(Cte);
 
+use crate::ast::ColumnDef;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct CteMutRec<T: AstInfo> {
+    pub name: Ident,
+    pub columns: Vec<ColumnDef<T>>,
```

Comment:
I think `ColumnDef<T>` here is probably not where we want to start, as it'll accept `NOT NULL` and `PRIMARY KEY` and all sorts of other weird constraints, IIRC. I think just `(Ident, T::DataType)` would be safer to start.

Reply:
We don't parse the `ColumnDef`, we parse just the names and types and put them in a `ColumnDef`. We can't use a pair because the various bits of autogen macro magic don't work on tuples.